### ISSUE TITLE
[java] AbstractClassWithoutAnyMethod: missing violation for nested classes

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#2390]
**Affects PMD Version:** 6.22.0

**Rule:** [AbstractClassWithoutAnyMethod](https://pmd.github.io/latest/pmd_rules_java_design.html#abstractclasswithoutanymethod)

**Description:**

Empty abstract classes are not detected, if they are nested within a non-empty class.
The same is true, if empty abstract classes are nested together with other non-empty (sibling) nested classes.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public void bar() { System.out.println("baz"); }
    private abstract class AbstractClass { }
}
```

**Expected outcome:**

The inner class "AbstractClass" should be flagged by this rule.

Note: This is fixed in #2384
